country-a-service/epps-service/src/main/java/dk/nsp/epps/service/PrescriptionService.java

```diff
@@ -109,24 +123,45 @@ public void submitDispensation(@NonNull String patientId, @NonNull Document disp
                     response),
                 caller);
         } catch (JAXBException e) {
-            // TODO: Cancel effectuation flow
+            // TODO: Cancel effectuation flow. (CFB 26/11/2024: Is this really necessary? We haven't done it, at it
+            //  does not seem to block anything. We should enquire at FMK as to what happens if we just begin the
+            //  effectuation, but never actually effectuate.)
             throw new CountryAException(HttpStatus.INTERNAL_SERVER_ERROR, "CreatePharmacyEffectuation failed", e);
+        } catch (MapperException e) {
+            throw new DataRequirementException(String.format("Could not create dispensation request due to mapper error: %s", e.getMessage()), e);
```

Comment:
I generalized this a bit, using the term dispensation about the CDA, and request about whatever we try to create with the CDA - giving us a somewhat standard error message of "Error mapping dispensation to request" followed by the mapper error.

To make sure that this is always true, I went through the calls to ensure that we only throw a mapper error on CDA mapping, and not on mapping of the response from FMK - which has been "promoted" to a Country A error (so that it naturally bubbles all the way to the OpenNCP.